### DISCUSSION TITLE
Problem with Waveshare ST7789 on a RP2040

### DISCUSSION THREAD

**Discussion by KBLiveSolutions** [Bodmer/TFT_eSPI#2052] [Q&A - General]
I have plugged [this Waveshare shield](https://www.waveshare.com/wiki/Pico-LCD-2) on a Pico, changed the User Settings like this :

`#define ST7789_DRIVER      // Full configuration option, define additional parameters below for this display

#define TFT_WIDTH  240 // ST7789 240 x 240 and 240 x 320
#define TFT_HEIGHT 320 // ST7789 240 x 320

#define TFT_BL   13            // LED back-light control pin
#define TFT_BACKLIGHT_ON HIGH  // Level to turn ON back-light (HIGH or LOW)
#define TFT_MISO -1
#define TFT_MOSI  11
#define TFT_SCLK  10
#define TFT_CS   9  // Chip select control pin
#define TFT_DC   8  // Data Command control pin
#define TFT_RST  12  // Reset pin (could connect to Arduino RESET pin)

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
//#define LOAD_FONT8N // Font 8. Alternative to Font 8 above, slightly narrower, so 3 digits fit a 160 pixel TFT
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

// Comment out the #define below to stop the SPIFFS filing system and smooth font code being loaded
// this will save ~20kbytes of FLASH
#define SMOOTH_FONT

#define SPI_FREQUENCY   5000000

// Optional reduced SPI frequency for reading TFT
#define SPI_READ_FREQUENCY  20000000

// The XPT2046 requires a lower SPI clock rate of 2.5MHz so we define that here:
#define SPI_TOUCH_FREQUENCY  2500000`

And Loading TFT_graphictest_one_lib with Earle's Philhower port leaves me with a black screen (backlight turns on and off by changing TFT_BACKLIGHT_ON).
I have tested the display with the provided micropython and C code, and it works fine...
I don't know what to do...

Here is the Read_user_setup

`TFT_eSPI ver = 2.4.76
Processor    = RP2040
Transactions = Yes
Interface    = SPI
Display driver = 7789
Display width  = 240
Display height = 320

MOSI    = GPIO 11
SCK     = GPIO 10
TFT_CS   = GPIO 9
TFT_DC   = GPIO 8
TFT_RST  = GPIO 12

TFT_BL           = GPIO 13
TFT_BACKLIGHT_ON = HIGH

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 5.00`

**Comment by joecampbell** [comment]
Try adding:
```
#define TFT_SPI_PORT 1
```

**Reply by KBLiveSolutions** [reply]
Thanks. I've already tried that with no success.
Also tried 2 other displays (Waveshare 4'' ILI9486 and Kuman palallel 8 bit ILI9488) with no luck either.
My Pico board works fine with other projects, the displays work fine on a Uno board... I really feel cursed.

I don't know what can possibly be wrong.

**Reply by joecampbell** [reply]
I had difficulty with the Waveshare PICO 1.3 ( https://www.waveshare.com/pico-lcd-1.3.htm ).
After a great deal of futzing about my working User_Setup.h is:
```
#define ST7789_DRIVER      // Full configuration option, define additional parameters below for this display

#define TFT_WIDTH  240
#define TFT_HEIGHT 240

#define TFT_SPI_PORT 1

// Pins - the PIO will control MOSI, SCLK and DC pins
// Any digital GPIO pins may be used
// #define TFT_MISO  -1 // MISO is not used or supported
#define TFT_MOSI  11
#define TFT_SCLK  10
#define TFT_CS     9  // Chip select control pin
#define TFT_DC     8  // Data Command control pin
#define TFT_RST   12  // Reset pin (could connect to Arduino RESET pin
#define TFT_BL    13  // LED back-light
#define TFT_BACKLIGHT_ON HIGH

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define SPI_FREQUENCY  20000000
```

This is running using the Arduino MBORD OS RP2040 Boards

I tried several examples plus my own code and it is working.

The example you mentioned `TFT_graphictest_one_lib` gives me a blank screen as well. Other examples (TFT_Clock, Orrery, Colour_Test, Boing_ball, etc.) seem to work.
